doc/guides/index.rst

```diff
@@ -32,3 +32,4 @@ User and Developer Guides
    west/index
    optimizations/index
    zephyr_cmake_package.rst
+   platformio/index.rst
```

Comment:
These are still mostly in alphabetical order; I think we should try to keep it that way by putting this with the p's.